src/modules/household/components/HouseholdMemberTable.tsx

```diff
@@ -8,97 +9,196 @@ import HouseholdActionButtons from './elements/HouseholdActionButtons';
 import HmisEnum from '@/modules/hmis/components/HmisEnum';
 import HohIndicator from '@/modules/hmis/components/HohIndicator';
 import { parseAndFormatDate } from '@/modules/hmis/hmisUtil';
 import { ClientPermissionsFilter } from '@/modules/permissions/PermissionsFilters';
-import { ClientDashboardRoutes } from '@/routes/routes';
+import {
+  ClientDashboardRoutes,
+  EnrollmentDashboardRoutes,
+} from '@/routes/routes';
 import { HmisEnums } from '@/types/gqlEnums';
 import { HouseholdClientFieldsFragment } from '@/types/gqlTypes';
 import generateSafePath from '@/utils/generateSafePath';
 
+export const nameColumnConfig = (currentClientId: string) => {
+  return {
+    header: 'Name',
+    key: 'name',
+    render: (h: HouseholdClientFieldsFragment) => {
+      const isCurrentClient = h.client.id === currentClientId;
+      const viewEnrollmentPath = generateSafePath(
+        EnrollmentDashboardRoutes.ENROLLMENT_OVERVIEW,
+        {
+          clientId: h.client.id,
+          enrollmentId: h.enrollment.id,
+        }
+      );
+      const routerLinkProps = isCurrentClient
+        ? undefined
+        : {
+            to: viewEnrollmentPath,
+            target: '_blank',
+          };
+
+      return (
+        <ClientName
+          client={h.client}
+          routerLinkProps={routerLinkProps}
+          bold={isCurrentClient}
+        />
+      );
+    },
+  };
+};
+
+export const HOUSEHOLD_MEMBER_COLUMNS = {
+  hohIndicator: {
+    header: '',
+    key: 'indicator',
+    width: '1%',
+    render: (hc: HouseholdClientFieldsFragment) => (
+      <HohIndicator relationshipToHoh={hc.relationshipToHoH} />
+    ),
+  },
+  clientName: ({
+    currentClientId,
+    linkToProfile = false,
+  }: {
+    currentClientId?: string;
+    linkToProfile?: boolean;
+  }) => ({
+    header: 'Name',
+    key: 'name',
+    render: (h: HouseholdClientFieldsFragment) => {
+      const isCurrentClient = h.client.id === currentClientId;
+      const linkTo = linkToProfile
+        ? generateSafePath(ClientDashboardRoutes.PROFILE, {
+            clientId: h.client.id,
+          })
+        : generateSafePath(EnrollmentDashboardRoutes.ENROLLMENT_OVERVIEW, {
+            clientId: h.client.id,
+            enrollmentId: h.enrollment.id,
+          });
+
+      return (
+        <ClientName
+          client={h.client}
+          routerLinkProps={isCurrentClient ? undefined : { to: linkTo }}
+          bold={isCurrentClient}
+        />
+      );
+    },
+  }),
+  enrollmentStatus: {
+    header: 'Status',
+    render: (hc: HouseholdClientFieldsFragment) => (
+      <EnrollmentStatus enrollment={hc.enrollment} />
+    ),
+  },
+  entryDate: {
+    header: 'Entry Date',
+    render: (hc: HouseholdClientFieldsFragment) =>
+      parseAndFormatDate(hc.enrollment.entryDate),
+  },
+  exitDate: (householdMembers: HouseholdClientFieldsFragment[]) => ({
+    header: 'Exit Date',
+    hide: !some(householdMembers, (m) => m.enrollment.exitDate),
```

Comment:
native js over lodash?
```suggestion
    hide: !householdMembers.some((m) => m.enrollment.exitDate),
```